File: Java/OperatingSystemFamilies.java
//package EnumerationOperatingSystem;
import java.util.*;

public enum OperatingSystemFamilies{
    
    WINDOWS ("WINDOWS", OperatingSystemClasses.DESKTOP),
	WINDOWSSERVER ("WINDOWS SERVER", OperatingSystemClasses.SERVER),
	WINDOWSMOBILE ("WINDOWS Mobile", OperatingSystemClasses.EMBEDDED),
	WINDOWSPHONE ("WINDOWS Phone", OperatingSystemClasses.EMBEDDED),
	WINDOWSTENMOBILE ("WINDOWS 10 Mobile", OperatingSystemClasses.EMBEDDED),
	UBUNTU ("Ubuntu", OperatingSystemClasses.DESKTOP),
	UBUNTUSERVER ("Ubuntu Server", OperatingSystemClasses.SERVER),
	LINUX ("Linux", OperatingSystemClasses.DESKTOP),
	LINUXSERVER ("Linux Server", OperatingSystemClasses.SERVER),
	MAC ("OS X", OperatingSystemClasses.DESKTOP),
	MACSERVER ("OS X Server", OperatingSystemClasses.SERVER),
	IOS ("iOS", OperatingSystemClasses.EMBEDDED),
	TVOS ("tvOS", OperatingSystemClasses.EMBEDDED),
	WATCHOS ("watchOS", OperatingSystemClasses.EMBEDDED),
	ANDROID ("Android", OperatingSystemClasses.EMBEDDED);

    private String familyName;
    private OperatingSystemClasses oClasses;

    OperatingSystemFamilies(String familyName, OperatingSystemClasses oClasses){
        this.familyName = familyName;
        this.oClasses = oClasses;
    }

    public List<OperatingSystemFamilies> GetAllFamiliesToClass(OperatingSystemClasses oClassToSearch){
        List<OperatingSystemFamilies> oReturnList = new ArrayList<OperatingSystemFamilies>();
        for (OperatingSystemFamilies OSF: OperatingSystemFamilies.values())
        {
            if (OSF.GetOClasses() == oClassToSearch){oReturnList.add(OSF);}
        }
        return oReturnList;
    }

    public OperatingSystemFamilies parse(String stringToParse){
        List<OperatingSystemFamilies> oList = GetAllFamiliesToClass(OperatingSystemClasses.DESKTOP.parse(stringToParse));
        for (OperatingSystemFamilies oItem:oList)
        {
            int partsOfStringFound = 0;
            for (String part:oItem.getFamilyName().split(" "))
            {
                if (stringToParse.toLowerCase().contains(part.toLowerCase())){partsOfStringFound+=1;}
            }
            if (partsOfStringFound==oItem.getFamilyName().split(" ").length){return oItem;}
        }
        return null;
    }

    public String getFamilyName(){
        return this.familyName;
    }

    public OperatingSystemClasses GetOClasses(){
        return this.oClasses;
    }
}

File: Python/OperatingSystemFamilies.py
from enum import Enum
import OperatingSystemClasses
import re

class OperatingSystemFamilies(Enum):

	WINDOWS = ("WINDOWS", OperatingSystemClasses.OperatingSystemClasses.DESKTOP)
	WINDOWSSERVER = ("WINDOWS SERVER", OperatingSystemClasses.OperatingSystemClasses.SERVER)
	WINDOWSMOBILE = ("WINDOWS Mobile", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)
	WINDOWSPHONE = ("WINDOWS Phone", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)
	WINDOWSTENMOBILE = ("WINDOWS 10 Mobile", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)
	UBUNTU = ("Ubuntu", OperatingSystemClasses.OperatingSystemClasses.DESKTOP)
	UBUNTUSERVER = ("Ubuntu Server", OperatingSystemClasses.OperatingSystemClasses.SERVER)
	LINUX = ("Linux", OperatingSystemClasses.OperatingSystemClasses.DESKTOP)
	LINUXSERVER =  ("Linux Server", OperatingSystemClasses.OperatingSystemClasses.SERVER)
	MAC = ("OS X", OperatingSystemClasses.OperatingSystemClasses.DESKTOP)
	MACSERVER = ("OS X Server", OperatingSystemClasses.OperatingSystemClasses.SERVER)
	IOS = ("iOS", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)
	TVOS = ("tvOS", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)
	WATCHOS = ("watchOS", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)
	ANDROID = ("Android", OperatingSystemClasses.OperatingSystemClasses.EMBEDDED)

	def __init__(self, family, osClass):
		self.osclass = osClass
		self.family = family

	def getAllClasses(self, classes):
		ret = []
		if classes == None:
			return OperatingSystemFamilies
		for i in OperatingSystemFamilies:
			if i.osclass == classes:
				ret.append(i)
		return ret

	def parse(self, string):
    	oClassInStr = OperatingSystemClasses.OperatingSystemClasses.DESKTOP.parse(string)
		lst = self.getAllClasses(oClassInStr)
		#lst = selfw.getAllClasses(OperatingSystemClasses.OperatingSystemClasses.DESKTOP.parse(string))
		iLengthOfObjectName = 0 
		for i in lst:
    		if len(i.family) > 0:
    			if (string.lower().find(i.family.lower())) >= 0:
    				oFamilyFound = i
					iLengthOfObjectName = len(i.family)
			#a = 0
			#lst2 = i.family.split(' ')
			#for j in lst2:
			#	if (string.lower()).find(j.lower()) >= 0:
			#		a += 1
			#if a == len(lst2):
			#	return i


